pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java

```diff
@@ -92,6 +93,9 @@ public static void validate(Schema schema) {
                 "Exception in getting arguments for transform function '" + transformFunction + "' for column '"
                     + column + "'", e);
           }
+        } else if (fieldSpec.getFieldType().equals(FieldSpec.FieldType.TIME) && fieldSpec.getFieldType()
```

Comment:
Why putting this in the else? Also, this filter can never match because a field cannot be both TIME and DATE_TIME